shoop/xtheme/rendering.py

```diff
@@ -18,36 +18,42 @@
 from shoop.xtheme.view_config import ViewConfig
 
 
-def get_view_config(context):
+def get_view_config(context, global_type=False):
     """
     Get a view configuration object for a Jinja2 rendering context.
 
     :param context: Rendering context
     :type context: jinja2.runtime.Context
+    :param global_type: Boolean indicating whether this is a global type
+    :type global_type: bool|False
     :return: View config
     :rtype: shoop.xtheme.view_config.ViewConfig
     """
     # This uses the Jinja context's technically-immutable vars dict
     # to cache the view configuration. This is fine in our case, I'd say.
+    # If global types do not match, must refetch view config
     request = context.get("request")
     config = context.vars.get("_xtheme_view_config")
-    if config is None:
+    if (config is None) or (global_type is not config.global_type):
         view_object = context.get("view")
         if view_object:
             view_class = view_object.__class__
             view_name = view_class.__name__
         else:
             view_name = "UnknownView"
+        if global_type:
+            view_name = "_XthemeGlobalView"
```

Comment:
Check my other comment on ViewConfig. Anyways this should be XTHEME_GLOBAL_VIEW_NAME instead of string. But I really do think that the view_name should be overriden in the ViewConfig if the global_type is True.